feat(java): Add automatic and custom performance metrics doc (#5604)
getsentry/sentry-docs: src/platform-includes/performance/automatic-performance-metrics/android.mdx

> If configured, the Android SDK can automatically collect the following performance metrics:

Author: Thanks for your suggestion! Yes, the SDK automatically captures those metrics by default. There are options to disable this but it's on by default.

> * Cold and Warm app start time - learn more in [App Start Instrumentation](/platforms/android/performance/instrumentation/automatic-instrumentation/#app-start-instrumentation).

Reviewer: `l`:
```suggestion
* Cold and warm app start time - learn more in [App Start Instrumentation](/platforms/android/performance/instrumentation/automatic-instrumentation/#app-start-instrumentation).
```

> * Cold and Warm app start time, see [App Start instrumentation](/platforms/android/performance/instrumentation/automatic-instrumentation/#app-start-instrumentation).
> * Frozen and slow frame rendering, details can be found on the [Slow and Frozen Frames page](/platforms/android/performance/instrumentation/automatic-instrumentation/#slow-and-frozen-frames).

Reviewer 2: ```suggestion
* Cold and Warm app start time - learn more in [App Start Instrumentation](/platforms/android/performance/instrumentation/automatic-instrumentation/#app-start-instrumentation).
* Frozen and slow frame rendering - learn more in [Slow and Frozen Frames](/platforms/android/performance/instrumentation/automatic-instrumentation/#slow-and-frozen-frames).
```
